src/test/java/io/jenkins/plugins/analysis/warnings/ReferenceFinderITest.java

```diff
@@ -0,0 +1,667 @@
+        enableWarnings(project, publisher -> {
+            publisher.setUnstableNewAll(3);
+            publisher.setReferenceJobName(refName);
+            publisher.setOverallResultMustBeSuccess(true);
+            publisher.setEnabledForFailure(true);
+        });
+
+        AnalysisResult result;
+
+        result = scheduleBuildAndAssertStatus(refJob, Result.SUCCESS);
+        assertThat(result).hasTotalSize(4);
+        assertThat(result).hasOverallResult(Result.SUCCESS);
+
+        refJob.getSomeWorkspace().deleteContents();
+        copyFilesToWorkspace(refJob, "eclipse2Warnings.txt", "checkstyle.xml");
+        issuesRecorder.setUnstableNewAll(3);
+        Shell shell = new Shell("exit 1");
+        refJob.getBuildersList().add(shell);
+
+        result = scheduleBuildAndAssertStatus(refJob, Result.FAILURE);
+        assertThat(result).hasTotalSize(2);
+        assertThat(result).hasOverallResult(Result.SUCCESS);
+        refJob.getBuildersList().remove(shell);
+
+        result = scheduleBuildAndAssertStatus(project, Result.UNSTABLE);
+        assertThat(result).hasTotalSize(6);
+        assertThat(result).hasOverallResult(Result.UNSTABLE);
+    }
+
+    @Test
+    public void shouldCreateSuccessResultWithOverAllMustNotBeSuccessWithReferenceBuild() throws IOException, InterruptedException {
+        FreeStyleProject refJob = createJobWithWorkspaceFile(refName, "eclipse2Warnings.txt");
+        IssuesRecorder issuesRecorder = enableWarnings(refJob, publisher -> {
+            publisher.setOverallResultMustBeSuccess(false);
+            publisher.setEnabledForFailure(true);
+            publisher.setUnstableNewAll(3);
+        });
+
+        FreeStyleProject project = createJobWithWorkspaceFile(jobName, "eclipse6Warnings.txt");
+        enableWarnings(project, publisher -> {
+            publisher.setUnstableNewAll(3);
+            publisher.setReferenceJobName(refName);
+            publisher.setOverallResultMustBeSuccess(true);
+            publisher.setEnabledForFailure(true);
+        });
+
+        AnalysisResult result;
+
+        result = scheduleBuildAndAssertStatus(refJob, Result.SUCCESS);
+        assertThat(result).hasTotalSize(2);
+        assertThat(result).hasOverallResult(Result.SUCCESS);
+
+        refJob.getSomeWorkspace().deleteContents();
+        copyFilesToWorkspace(refJob, "eclipse4Warnings.txt", "checkstyle.xml");
+
+        Shell shell = new Shell("exit 1");
+        refJob.getBuildersList().add(shell);
+
+        result = scheduleBuildAndAssertStatus(refJob, Result.FAILURE);
+        assertThat(result).hasTotalSize(4);
+        assertThat(result).hasOverallResult(Result.SUCCESS);
+        refJob.getBuildersList().remove(shell);
+
+        result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+        assertThat(result).hasTotalSize(6);
+        assertThat(result).hasOverallResult(Result.SUCCESS);
+    }
+
+    // Copied from IssuesRecorderITest
```

Comment:
I´ve copied these from a different test class. Should these be refactored to (maybe) a ITestUtils class?

Reply:
I'll do that on my own when all tests are merged... I think some rework is required to consolidate the different classes anyway.